### ISSUE TITLE
`pip-compile-multi`: ImportError: cannot import name 'DEV_PKGS' from 'pip._internal.commands.freeze'

### ISSUE THREAD

**Issue by ahal** [taskcluster/taskgraph#364]
We're currently unable to update dependencies with latest `pip-compile-multi`. I reproduced both with the `pin-helper.sh` script as well as locally.

Here's the full traceback:
```
$ pip-compile-multi -g base -g test -g dev -g docs
Locking requirements/base.in to requirements/base.txt. References: []
ERROR executing /home/ahal/.pyenv/versions/3.7.17/envs/taskgraph_37/bin/python3.7 -m piptools compile --no-header --verbose --generate-hashes --no-emit-index-url --resolver=legacy --emit-trusted-host --upgrade --rebuild --output-file requirements/base.txt requirements/base.in
Exit code: 1
Traceback (most recent call last):
  File "/home/ahal/.pyenv/versions/3.7.17/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/ahal/.pyenv/versions/3.7.17/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/ahal/.pyenv/versions/3.7.17/envs/taskgraph_37/lib/python3.7/site-packages/piptools/__main__.py", line 5, in <module>
    from piptools.scripts import compile, sync
  File "/home/ahal/.pyenv/versions/3.7.17/envs/taskgraph_37/lib/python3.7/site-packages/piptools/scripts/sync.py", line 17, in <module>
    from .. import sync
  File "/home/ahal/.pyenv/versions/3.7.17/envs/taskgraph_37/lib/python3.7/site-packages/piptools/sync.py", line 11, in <module>
    from pip._internal.commands.freeze import DEV_PKGS
ImportError: cannot import name 'DEV_PKGS' from 'pip._internal.commands.freeze' (/home/ahal/.pyenv/versions/3.7.17/envs/taskgraph_37/lib/python3.7/site-packages/pip/_internal/commands/freeze.py)
```

**Comment by ahal** [comment]
This is caused by a backwards incompatible change in `pip` that broke `pip-tools`. The issue has since been fixed in `pip-tools`, but they dropped support for Python 3.7 awhile back and the fix isn't in any released versions for 3.7.

I guess the best workaround here is to downgrade to pip<23.2. Hopefully we can drop support for Python 3.7 soon.

**Comment by ahal** [comment]
Gecko is on the verge of dropping 3.7 support, so we can fix this by doing the same here.

**Comment by ahal** [comment]
Fixed by dropping Python 3.7 support.
